### PR TITLE
Add X-Robots-Tag header to stop crawlers from indexing media

### PR DESCRIPTION
**Pull request by aaronraimist**
Fixes / related to: https://github.com/matrix-org/synapse/issues/6533

This should do essentially the same thing as a robots.txt file telling robots to not index the media repo. https://developers.google.com/search/reference/robots_meta_tag

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

File: changelog.d/8887.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Add X-Robots-Tag header to stop web crawlers from indexing media.
```
Comment (anoadragon453) [inline changelog.d/8887.bugfix:1]:
```suggestion
Add `X-Robots-Tag` header to stop web crawlers from indexing media.
```

File: synapse/rest/media/v1/_base.py [hunk 2]
```diff
@@ -155,6 +155,8 @@ def _quote(x):
     request.setHeader(b"Cache-Control", b"public,max-age=86400,s-maxage=86400")
     request.setHeader(b"Content-Length", b"%d" % (file_size,))
 
+    request.setHeader(b"X-Robots-Tag", "noindex, nofollow, noarchive, noimageindex")
```
Comment (anoadragon453) [inline synapse/rest/media/v1/_base.py:158]: Could you add a comment here explaining what this header's purpose is?

File: changelog.d/8887.bugfix (hunk 1 above)
Comment (anoadragon453) [inline changelog.d/8887.bugfix:1]: I think this is more a change (`.misc`) rather than fixing a bug.

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
It'd be nice to add a test to confirm the header is appearing in request responses.

Also note that `X-Robots-Tag` has not yet been standardized according to MDN: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers.

File: changelog.d/8887.bugfix (hunk 1 above)
Comment (clokep) [reply changelog.d/8887.bugfix:1]: This is probably fairly user visible so shouldn't be `misc` IMO.

Comment (anoadragon453) [reply changelog.d/8887.bugfix:1]: Hm, good point.

Comment (aaronraimist) [reply changelog.d/8887.bugfix:1]: Could also be a feature

**Comment by aaronraimist** [comment]
Okay I've added a test. Let me know if that looks correct, I'm not really sure what I am doing.

File: tests/rest/media/v1/test_media_storage.py [hunk 3]
```diff
@@ -362,3 +362,12 @@ def _test_thumbnail(self, method, expected_body, expected_found):
                     "error": "Not found [b'example.com', b'12345']",
                 },
             )
+
+    def test_x_robots_tag_header(self):
```
Comment (anoadragon453) [inline tests/rest/media/v1/test_media_storage.py:366]:
```suggestion
    def test_x_robots_tag_header(self):
    """
    Tests that the `X-Robots-Tag` is present, which informs web crawlers to not index, archive, or
    follow links in media.
    """
```

I'd just add this comment in again to give some explanation about why we want to see this.

File: changelog.d/8887.bugfix [hunk 4]
```diff
@@ -0,0 +1 @@
+Add `X-Robots-Tag` header to stop web crawlers from indexing media.
```
Comment (anoadragon453) [inline changelog.d/8887.bugfix:1]: Oops, I marked the earlier discussion here as resolved too early. Could you change this to a `.feature` file please?

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
Test looks good, thanks! Just some minor nits and then lgtm.

**Review by anoadragon453 — APPROVED** [review]
LGTM, thanks!

CI failures are unrelated to this change.

**Pull request merged**